src/BorderBox.js

```diff
@@ -1,18 +1,30 @@
 import {withSystemProps} from './system-props'
+import {borders, borderColor, fontSize, borderRadius, boxShadow} from 'styled-system'
+import styled from 'styled-components'
+import theme, {colors} from './theme'
 import Box from './Box'
 
-const BorderBox = withSystemProps(
-  {
-    is: Box,
-    border: 1,
-    borderColor: 'gray.2',
-    borderRadius: 1
-  },
-  ['boxShadow']
-)
+const BorderBox = styled(Box)`
+ ${borders}
+ ${borderColor}
+ ${borderRadius}
+ ${boxShadow}
+`
 
+// set default props in defaultProps instead of in call to withSystemProps
+BorderBox.defaultProps = {
+  border: theme.borders[1],
+  borderColor: colors.gray[2],
+  borderRadius: theme.radii[1],
+  theme // set our theme to the default if no theme is provided
+}
```

Comment:
These default props won't update if the consumer is using a custom `theme`, right? Is that a problem?

Reply:
@shawnbot this is setting the default theme here only if the user doesn't provide one, or are you talking about something else?

@colebemis - yeah, these values won't use the provided theme value if a theme is provided, it always defaults to our theme values... though I'm not sure if that's a huge problem or not? IMO the default values _should_ be set to values from the Primer theme, and can always be overridden if you'd like to provide your own theme by providing a value for that prop. cc @broccolini

Reply:
Ah right we'd have to do it like this if we want to use `themeGet` since props aren't available outside the context of the component:
https://github.com/primer/components/pull/362/commits/7e1c2c1f9b52fecbcec8057db379e4e00c12a74b

Reply:
meeeh this is sort of annoying though because in that commit, if the user doesn't provide a custom theme, `themeGet` here doesn't recognize the theme we've provided in defaultProps (defaultProps is removed from the above commit but I just tested locally and confirmed :/)

Reply:
fwiw this isn't working currently in primer/components anyways 😂

Reply:
I discovered something interesting... the reason this isn't working in primer/components (more accurately... the docs for primer/components) is because the `theme` prop for every component rendered in the docs is being overridden by the theme provided to `Layout` from `mdx-docs`... so the default is never set in any component because there is already a `theme` prop provided 🤦‍♀️ 

Two take-aways from this:
1) We should really have some sort of clean-ish dev environment so we aren't deciphering between primer/components issues and issues due to clashes with our docs stack 😂 

2) I'd be keen to pair with you @shawnbot tomorrow to pull mdx-docs out and use your system for primer.style/design so that we're on the same page between repos and don't need to worry about theme clashes with `Layout`.

Reply:
> Ah right we'd have to do it like this if we want to use `themeGet` since props aren't available outside the context of the component: [7e1c2c1](https://github.com/primer/components/commit/7e1c2c1f9b52fecbcec8057db379e4e00c12a74b)
> 
> Doesn't feel _too_ bad though?

@emplums and team, I read through this RFC PR, dig it. I apologize if you arrived at a conclusion or answer already regarding the `themeGet` utility, however, in the off chance that you're still exploring approaches, I have found that you can still leverage `themeGet` successfully in your custom Primer component when the theme provider is not defined, but the default `theme` prop is, as follows (ala styled-components)
```suggestion
border: ${props => themeGet('borders.1'), theme.borders[1])(props)};
```
Original version: `border: ${themeGet('borders.1'), theme.borders[1]};` from [7e1c2c1](https://github.com/primer/components/commit/7e1c2c1f9b52fecbcec8057db379e4e00c12a74b)

Reply:
@mannybecerra ah yes! We found a bug that was preventing the `defaultProps` from kicking in, which is why we were having issues with `themeGet`. We've resolved that and now it should work swimmingly! 🙌